embassy-stm32/src/rcc/f013.rs

```diff
@@ -187,21 +234,39 @@ pub(crate) unsafe fn init(config: Config) {
                 }
                 (Pllsrc::HSI_DIV2, unwrap!(hsi))
             }
+            #[cfg(not(stm32f107))]
             PllSource::HSE => (Pllsrc::HSE_DIV_PREDIV, unwrap!(hse)),
+            #[cfg(stm32f107)]
+            PllSource::HSE => (
+                Pllsrc::HSE_DIV_PREDIV,
+                match prediv1_src {
+                    PreDiv1Src::HSE => unwrap!(hse),
+                    PreDiv1Src::PLL2 => unwrap!(pll2freq),
+                },
+            ),
             #[cfg(rcc_f0v4)]
             PllSource::HSI48 => (Pllsrc::HSI48_DIV_PREDIV, unwrap!(hsi48)),
         };
+        #[cfg(not(stm32f107))]
         let in_freq = src_freq / pll.prediv;
+
+        #[cfg(stm32f107)]
+        let in_freq = src_freq / (pll.prediv.to_bits() + 1);
```

Comment:
this shouldn't be needed, `/ pll.prediv` does the right thing automatically (it's a Div impl generated in build.rs, it's *not* dividing by the bit value)

Reply:
okay, for this to work add `Prediv1`, `Prediv2` here. https://github.com/embassy-rs/embassy/blob/010d4622f988de813ecddf69b84273f72ab122cf/embassy-stm32/build.rs#L1386 . Then you can directly divide instead of having to do `.to_bits()`.